src/menus/LabelMenus.cpp

```diff
@@ -591,11 +626,15 @@ void OnJoinLabels(const CommandContext &context)
    if (selectedRegion.isPoint())
       return;
 
-   auto editfunc = [&](Track &track, double t0, double t1) {
+   auto editfunc = [&](Track& track, double t0, double t1, ProgressReporter reportProgress) {
       assert(track.IsLeader());
-      track.TypeSwitch( [&](WaveTrack &t) { t.Join(t0, t1); } );
+      track.TypeSwitch(
+         [&](WaveTrack& t) { t.Join(t0, t1, std::move(reportProgress)); });
    };
-   EditByLabel(project, tracks, selectedRegion, editfunc);
+   EditByLabel(
+      project, tracks, selectedRegion, editfunc,
+      BasicUI::MakeProgress(
+         XO("Pre-processing"), XO("Rendering Time-Stretched Audio")));
```

Comment:
Strings.  @LWinterberg  ?

Reply:
Part of the utility functions proposed in recent commits.